### ISSUE TITLE
fairseq OOM error when training on multiple GPU

### ISSUE THREAD

**Issue by marwagaser** [pytorch/fairseq#3885]
I am following the [mBART fine-tuning example](https://github.com/pytorch/fairseq/tree/master/examples/mbart); however, when I try to train on multiple GPUs (two or more) I get an OOM error. Nevertheless, when I train on 1 GPU, it works fine

I followed this issue https://github.com/pytorch/fairseq/issues/2867 and changed --ddp-backend=no_c10d and decreased the update_freq = 1. However, I am still getting the error:  `attempting to recover from OOM in forward/backward pass`.

Here is the full command that I used as per the mBART example:

`python3 ${FAIRSEQ}/train.py ${DATADIR}   --encoder-normalize-before --decoder-normalize-before --arch mbart_large --layernorm-embedding   --task translation_from_pretrained_bart   --source-lang en_XX --target-lang ja_XX   --criterion label_smoothed_cross_entropy --label-smoothing 0.2   --optimizer adam --adam-eps 1e-06 --adam-betas '(0.9, 0.98)'   --lr-scheduler polynomial_decay --lr 3e-05 --warmup-updates 2500 --total-num-update 40000   --dropout 0.3 --attention-dropout 0.1 --weight-decay 0.0   --max-tokens 1024 --update-freq 1   --save-interval 1 --save-interval-updates 5000 --keep-interval-updates 10 --no-epoch-checkpoints   --seed 222 --log-format simple --log-interval 2 --reset-optimizer --reset-meters --reset-dataloader --reset-lr-scheduler   --langs $langs   --ddp-backend no_c10d`

#### What's your environment?

 - fairseq Version (master):
 - PyTorch Version (1.9.0+cu102)
 - OS (e.g., Fedora):
 - How you installed fairseq (source):
 - Python version: 3.9.7
 - CUDA/cuDNN version: cu102
 - GPU models and configuration: NVIDIA-SMI 470.63.01    Driver Version: 470.63.01    CUDA Version: 11.4

**Comment by kurtisxx** [comment]
I confirm that I'm getting the same warning when I'm trying to train a simultaneous translation model (MMA-Hard) from scratch on a node with 4 GPUs.

`fairseq-train ~/wmt15_de_en_data/new_data_bin/wmt15.de-en/ --simul-type hard_aligned --mass-preservation --criterion latency_augmented_label_smoothed_cross_entropy --latency-var-weight 0.1 --max-update 50000 --arch transformer_monotonic_vaswani_wmt_en_de_big --optimizer adam --adam-betas '(0.9, 0.98)' --lr-scheduler 'inverse_sqrt' --warmup-init-lr 1e-7  --warmup-updates 4000 --lr 5e-4 --stop-min-lr 1e-9 --clip-norm 0.0 --weight-decay 0.0001 --dropout 0.3 --label-smoothing 0.1 --max-tokens 3584 --left-pad-source --update-freq 16
`

Since it was a warning, it looks like training  is continuing. Is this something to worry about?
